### PR TITLE
Remove versions from dev dependencies

### PR DESCRIPTION
**Pull request by maackle**
### Summary

Hypothesis: There's no need for versions in dev dependencies: they only create opportunities for forgetting to update one, or missing one in a merge conflict, leading to a confusing or annoying error.

### TODO:
- [ ] ~~CHANGELOG(s) updated with appropriate info~~
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

**Review by steveeJ — APPROVED** [review]
agreed, thanks!

File: crates/kitsune_p2p/kitsune_p2p/Cargo.toml [hunk 1]
```diff
@@ -51,8 +51,9 @@ mockall = { version = "0.10.2", optional = true }
 [dev-dependencies]
 kitsune_p2p_bootstrap = { path = "../bootstrap" }
 contrafact = { version = "0.1.0-dev.1" }
-kitsune_p2p_timestamp = { version = "0.0.9", path = "../timestamp", features = ["now", "arbitrary"] }
-kitsune_p2p_types = { version = "0.0.24", path = "../types", features = ["test_utils"] }
+kitsune_p2p_bootstrap = { path = "../bootstrap" }
```
Comment (steveeJ) [inline crates/kitsune_p2p/kitsune_p2p/Cargo.toml:54]: seems to be a duplicate

```suggestion
```

File: crates/kitsune_p2p/kitsune_p2p/Cargo.toml [hunk 2]
```diff
@@ -51,8 +51,9 @@ mockall = { version = "0.10.2", optional = true }
 [dev-dependencies]
 kitsune_p2p_bootstrap = { path = "../bootstrap" }
```
Comment (maackle) [inline crates/kitsune_p2p/kitsune_p2p/Cargo.toml:52]:
```suggestion
```

**Comment by steveeJ** [comment]
CI failed with this:

```
Error: the following crates are blocked but required for the release: 

--------------------------------------------------------------------------------
DISALLOWED BLOCKING CRATES
Showing states: * Disallowed Blocking 
--------------------------------------------------------------------------------
holochain                     [HasWildcardDependency]
```

we should probably make it tolerate wildcard deps in dev-dependencies. i'll take a look at that and push the changes to this branch.

**Pull request merged**